plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -0,0 +1,651 @@
+      All TIME values must be between 00:00:00 and 23:59:59.999999999. TIME internally stores “wallclock” time, and all operations on TIME values are performed without taking any time zone into consideration.
+     */
+    private static final int SNOWFLAKE_MAX_SUPPORTED_TIMESTAMP_PRECISION = 9;
+    private static final Logger log = Logger.get(SnowflakeClient.class);
+    private static final DateTimeFormatter SNOWFLAKE_DATETIME_FORMATTER = DateTimeFormatter.ofPattern("y-MM-dd'T'HH:mm:ss.SSSSSSSSSXXX");
+    private static final DateTimeFormatter SNOWFLAKE_DATE_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd");
+    private static final DateTimeFormatter SNOWFLAKE_TIMESTAMP_FORMATTER = DateTimeFormatter.ofPattern("y-MM-dd'T'HH:mm:ss.SSSSSSSSS");
+    private static final DateTimeFormatter SNOWFLAKE_TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm:ss.SSSSSSSSS");
+    private final Type jsonType;
+    private final AggregateFunctionRewriter aggregateFunctionRewriter;
+
+    private interface WriteMappingFunction
+    {
+        WriteMapping convert(Type type);
+    }
+
+    private interface ColumnMappingFunction
+    {
+        Optional<ColumnMapping> convert(JdbcTypeHandle typeHandle);
+    }
+
+    private static final TimeZone UTC_TZ = TimeZone.getTimeZone(ZoneId.of("UTC"));
+    // Mappings for JDBC column types to internal Trino types
+    private static final Map<Integer, ColumnMapping> STANDARD_COLUMN_MAPPINGS = ImmutableMap.<Integer, ColumnMapping>builder()
+            .put(Types.BOOLEAN, StandardColumnMappings.booleanColumnMapping())
+            .put(Types.TINYINT, StandardColumnMappings.tinyintColumnMapping())
+            .put(Types.SMALLINT, StandardColumnMappings.smallintColumnMapping())
+            .put(Types.INTEGER, StandardColumnMappings.integerColumnMapping())
+            .put(Types.BIGINT, StandardColumnMappings.bigintColumnMapping())
+            .put(Types.REAL, StandardColumnMappings.realColumnMapping())
+            .put(Types.DOUBLE, StandardColumnMappings.doubleColumnMapping())
+            .put(Types.FLOAT, StandardColumnMappings.doubleColumnMapping())
+            .put(Types.BINARY, StandardColumnMappings.varbinaryColumnMapping())
+            .put(Types.VARBINARY, StandardColumnMappings.varbinaryColumnMapping())
+            .put(Types.LONGVARBINARY, StandardColumnMappings.varbinaryColumnMapping())
+            .buildOrThrow();
+
+    private static final Map<String, ColumnMappingFunction> SHOWFLAKE_COLUMN_MAPPINGS = ImmutableMap.<String, ColumnMappingFunction>builder()
+            .put("time", typeHandle -> {
+                //return Optional.of(columnMappingPushdown(timeColumnMapping(typeHandle)));
+                return Optional.of(timeColumnMapping(typeHandle));
+            })
+            .put("timestampntz", typeHandle -> {
+                return Optional.of(timestampColumnMapping(typeHandle));
```

Comment:
These can be lambda expression instead of blocks:

```java
.put("timestampntz", typeHandle -> Optional.of(timestampColumnMapping(typeHandle)))
```